pkg/generate/generate.go

```diff
@@ -47,6 +48,12 @@ func (c *Controller) processGR(gr *kyverno.GenerateRequest) error {
 	events := failedEvents(err, *gr, *resource)
 	c.eventGen.Add(events...)
 
+	// Need not update the stauts when policy doesn't apply on resource, because all the generate requests are removed by the cleanup controller
+	if err != nil && strings.Contains(err.Error(), "does not apply to resource") {
```

Comment:
would be best not to use error strings for comparison, as these can be changed. Instead we can introduce an error type and check for that. See: https://blog.golang.org/go1.13-errors.